### PR TITLE
DataForm: provide a better default for render when field has elements

### PR DESCRIPTION
**Pull request by oandregal**
Related https://github.com/WordPress/gutenberg/issues/55101
Follow-up to https://github.com/WordPress/gutenberg/pull/64299#discussion_r1706906095

## What?

Provide a better default `render` function when the field has declared `elements`.

## Why?

Note how the `author` and `status` fields are rendered in the before/after.

Before:

https://github.com/user-attachments/assets/35471e35-eb1b-4684-9aae-1ccc0bdddc73

After:

https://github.com/user-attachments/assets/7a99de35-263d-4409-86f7-072422aa64bb

## Testing Instructions

`npm install && npm run storybook:dev` and verify `author` and `status` are properly rendered.

### CODE REVIEW

File: packages/dataviews/src/normalize-fields.ts [hunk 1]
```diff
@@ -41,11 +41,22 @@ export function normalizeFields< Item >(
 
 		const Edit = field.Edit || fieldTypeDefinition.Edit;
 
+		const renderFromElements = ( { item }: { item: Item } ) => {
+			const value = getValue( { item } );
+			return (
+				field?.elements?.find( ( element ) => element.value === value )
+					?.label || getValue( { item } )
+			);
```
Comment (youknowriad) [inline packages/dataviews/src/normalize-fields.ts]: We may want to start adding some unit tests for this function

**Review by youknowriad — APPROVED** [review]

**Comment by oandregal** [comment]
The fix doesn't seem to work well with `author`: looking into it.

File: packages/dataviews/src/normalize-fields.ts [hunk 2]
```diff
@@ -43,10 +43,13 @@ export function normalizeFields< Item >(
...
 		const renderFromElements = ( { item }: { item: Item } ) => {
 			const value = getValue( { item } );
-			return (
-				field?.elements?.find( ( element ) => element.value === value )
-					?.label || getValue( { item } )
-			);
+			const label = field?.elements?.find( ( element ) => {
+				// Intentionally using == here to allow for type coercion.
+				// eslint-disable-next-line eqeqeq
+				return element.value == value;
```
Comment (youknowriad) [inline packages/dataviews/src/normalize-fields.ts:49]: Isn't the issue in the field though. If the field is "string" its values should be string both in `getValue` and `elements` for me. and same for integer no?

File: packages/dataviews/src/normalize-fields.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ export function normalizeFields< Item >(
 
 		const getValue =
 			field.getValue ||
-			( ( { item }: { item: ItemRecord } ) => item[ field.id ] );
+			( ( { item }: { item: Item } ) => item[ field.id as keyof Item ] );
```
Comment (youknowriad) [inline packages/dataviews/src/normalize-fields.ts:21]: Why did you change this?

Comment (oandregal) [reply packages/dataviews/src/normalize-fields.ts:21]: Sorry I didn't notice this before. I see this was addressed at https://github.com/WordPress/gutenberg/pull/64367

Comment (youknowriad) [reply packages/dataviews/src/normalize-fields.ts:21]: Yeah, not important right now :)

File: packages/dataviews/src/normalize-fields.ts (hunk 2 above)
Comment (oandregal) [reply packages/dataviews/src/normalize-fields.ts:49]: https://github.com/WordPress/gutenberg/pull/64391 may be a better approach.
